<launch>
    <arg name="l_serial_port" default="/dev/ttyUSB22" />
    <arg name="r_serial_port" default="/dev/ttyUSB23" />
    <arg name="l_joint_name" default="gripper_l_center_joint" />
    <arg name="r_joint_name" default="gripper_r_center_joint" />

	<node pkg="sensor_tools" name="serial_gripper_imu_l" type="serial_gripper_imu" output="screen" respawn="true">
	    <param name="serial_port" value="$(arg l_serial_port)"/>
        <param name="joint_name" value="$(arg l_joint_name)"/>
        <remap from="/imu/data" to="/imu_l/data"/>
        <remap from="/gripper/data" to="/gripper_l/data"/>
        <remap from="/gripper/ctrl" to="/gripper_l/ctrl"/>
        <remap from="/joint_state_ctrl" to="/joint_states_l"/>
        <remap from="/joint_state_info" to="/joint_states_single_l"/>
        <remap from="/joint_state_gripper" to="/joint_states_gripper_l"/>
        <remap from="/gripper/joint_state" to="/gripper_l/joint_state"/>
        <remap from="/teleop_trigger" to="/teleop_trigger_l"/>
	</node>
	<node pkg="sensor_tools" name="serial_gripper_imu_r" type="serial_gripper_imu" output="screen" respawn="true">
	    <param name="serial_port" value="$(arg r_serial_port)"/>
        <param name="joint_name" value="$(arg r_joint_name)"/>
        <remap from="/imu/data" to="/imu_r/data"/>
        <remap from="/gripper/data" to="/gripper_r/data"/>
        <remap from="/gripper/ctrl" to="/gripper_r/ctrl"/>
        <remap from="/joint_state_ctrl" to="/joint_states_r"/>
        <remap from="/joint_state_info" to="/joint_states_single_r"/>
        <remap from="/joint_state_gripper" to="/joint_states_gripper_r"/>
        <remap from="/gripper/joint_state" to="/gripper_r/joint_state"/>
        <remap from="/teleop_trigger" to="/teleop_trigger_r"/>
	</node>

    <include file="$(find piper)/launch/start_double_piper.launch"/>

    <include file="$(find pika_locator)/launch/pika_double_locator.launch"/>

    <node name="piper_IK_l" pkg="remote_operation" type="piper_ik.py" output="screen">
        <param name="index_name" value="_l" />
        <rosparam command="load" file="$(find remote_operation)/config/piper_params.yaml" />
        <!-- Parameters gripper_xyzrpy and target_joint_state loaded from YAML -->
    </node>

    <node name="piper_IK_r" pkg="remote_operation" type="piper_ik.py" output="screen" >
        <param name="index_name" value="_r" />
        <rosparam command="load" file="$(find remote_operation)/config/piper_params.yaml" />
        <!-- Parameters gripper_xyzrpy and target_joint_state loaded from YAML -->
    </node>

</launch>




  


